File: finalProject/finalProjectBackend/ParkplayFinal/src/main/java/org/example/parkplayfinal/model/Reservation.java
package org.example.parkplayfinal.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

/**Resprezinta o rezervare a unui loc de parcare dintr-o parcare pentru un user cu rol de driver*/
@Data
@NoArgsConstructor
public class Reservation {
    private Long id;
    private Long driverId;
    private Long spotId;
    private LocalDateTime startTime;
    private LocalDateTime endTime;

    public Reservation(Long driverId, Long spotId, LocalDateTime startTime, LocalDateTime endTime) {
        this.driverId = driverId;
        this.spotId = spotId;
        this.startTime = startTime;
        this.endTime = endTime;
    }
}

File: finalProjectFrontend/parkplay_frontend/lib/models/reservation.dart
class Reservation {
  final int id;
  final int driverId;
  final int spotId;
  final String startTime;
  final String endTime;

  Reservation({
    required this.id, 
    required this.driverId, 
    required this.spotId, 
    required this.startTime, 
    required this.endTime
  });

  factory Reservation.fromJson(Map<String, dynamic> json) {
    return Reservation(
      id: json['id'],
      driverId: json['driverId'],
      spotId: json['spotId'],
      startTime: json['startTime'],
      endTime: json['endTime'],
    );
  }
}